src/test/java/uk/gov/hmcts/dm/controller/StoredDocumentDeleteControllerTests.java

```diff
@@ -0,0 +1,68 @@
+package uk.gov.hmcts.dm.controller;
+
+import org.junit.Test;
+import uk.gov.hmcts.dm.commandobject.DeleteCaseDocumentsCommand;
+import uk.gov.hmcts.dm.componenttests.ComponentTestBase;
+import uk.gov.hmcts.dm.componenttests.TestUtil;
+import uk.gov.hmcts.dm.domain.StoredDocument;
+import uk.gov.hmcts.dm.response.CaseDocumentsDeletionResults;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.when;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+public class StoredDocumentDeleteControllerTests extends ComponentTestBase {
+
+    @Test
+    public void testValidDeleteCaseDocumentsCommand() throws Exception {
+        DeleteCaseDocumentsCommand deleteCaseDocumentsCommand = new DeleteCaseDocumentsCommand("0123456789123456");
+        List<StoredDocument> storedDocuments = Stream.of(TestUtil.STORED_DOCUMENT).collect(Collectors.toList());
+
+        CaseDocumentsDeletionResults caseDocumentsDeletionResults = new CaseDocumentsDeletionResults();
+        caseDocumentsDeletionResults.setCaseDocumentsFound(storedDocuments.size());
+        caseDocumentsDeletionResults.setMarkedForDeletion(storedDocuments.size());
+
+        when(
+            this.searchService
+                .findStoredDocumentsByCaseRef(eq(deleteCaseDocumentsCommand)))
+            .thenReturn(storedDocuments);
+
+        when(
+            this.auditedStoredDocumentOperationsService
+                .deleteCaseStoredDocuments(eq(storedDocuments)))
+            .thenReturn(caseDocumentsDeletionResults);
+
+        restActions
+            .withAuthorizedUser("userId")
+            .withAuthorizedService("ccd_data")
+            .post("/documents/delete", deleteCaseDocumentsCommand)
+            .andExpect(status().isOk());
+    }
+
+    @Test
+    public void testInValidDeleteCaseDocumentsCommand() throws Exception {
+        DeleteCaseDocumentsCommand deleteCaseDocumentsCommand = new DeleteCaseDocumentsCommand("123");
+
+        restActions
+            .withAuthorizedUser("userId")
+            .withAuthorizedService("ccd_data")
+            .post("/documents/delete", deleteCaseDocumentsCommand)
+            .andExpect(status().is4xxClientError());
+    }
+
+    @Test
+    public void testEmptyDeleteCaseDocumentsCommand() throws Exception {
+        DeleteCaseDocumentsCommand deleteCaseDocumentsCommand = new DeleteCaseDocumentsCommand();
+
+        restActions
+            .withAuthorizedUser("userId")
+            .withAuthorizedService("ccd_data")
+            .post("/documents/delete", deleteCaseDocumentsCommand)
+            .andExpect(status().is4xxClientError());
+    }
+
+}
```

Comment:
This seems to be Mocked Integration tests. We will need functional tests which will create required test data and then call different Delete scenarios. Something similar to what we have for other endpoints here - https://github.com/hmcts/document-management-store-app/tree/master/src/functionalTest/groovy/uk/gov/hmcts/dm/functional

Reply:
Added 8 functional tests under DeleteDocumentsIT.groovy.